src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -116,6 +118,19 @@ impl<Src: OpendalSource> OpendalReader<Src> {
             for chunk in chunk_stream {
                 yield chunk?;
             }
+
+            if let EncodingProperties::Parquet = &self.parser_config.specific.encoding_config {
+                // We no longer determine whether a file is finished by comparing `offset >= size`, the reader itself can sense whether it has reached the end.
+                // Therefore, after a file is read completely, we yield one more chunk marked as EOF, with its offset set to `usize::MAX` to indicate that it is finished.
+                // In fetch executor, if `offset = usize::MAX` is encountered, the corresponding file can be deleted from the state table.
```

Comment:
so for all encoding (parquet and json?), we have the usize::MAX as the EOF mark?

Reply:
yes, I think usize::MAX is safe.